clients/src/main/java/org/apache/kafka/clients/admin/DeleteTopicsResult.java

```diff
@@ -30,24 +31,56 @@
  */
 @InterfaceStability.Evolving
 public class DeleteTopicsResult {
-    final Map<String, KafkaFuture<Void>> futures;
+    private final Map<Uuid, KafkaFuture<Void>> topicIdFutures;
+    private final Map<String, KafkaFuture<Void>> nameFutures;
 
-    protected DeleteTopicsResult(Map<String, KafkaFuture<Void>> futures) {
-        this.futures = futures;
+    protected DeleteTopicsResult(Map<Uuid, KafkaFuture<Void>> topicIdFutures, Map<String, KafkaFuture<Void>> nameFutures) {
+        if (topicIdFutures != null && nameFutures != null)
+            throw new IllegalArgumentException("topicIdFutures and nameFutures can not both be specified.");
+        this.topicIdFutures = topicIdFutures;
+        this.nameFutures = nameFutures;
     }
 
+    protected static DeleteTopicsResult ofTopicIds(Map<Uuid, KafkaFuture<Void>> topicIdFutures) {
+        DeleteTopicsResult result = new DeleteTopicsResult(topicIdFutures, null);
+        return result;
+    }
+
+    protected static DeleteTopicsResult ofTopicNames(Map<String, KafkaFuture<Void>> nameFutures) {
+        DeleteTopicsResult result = new DeleteTopicsResult(null, nameFutures);
+        return result;
+    }
+
+    /**
+     * @return a map from topic IDs to futures which can be used to check the status of
+     * individual deletions if the deleteTopics request used topic IDs. Otherwise return null.
+     */
+    public Map<Uuid, KafkaFuture<Void>> topicIdValues() {
+        return topicIdFutures;
+    }
+
+    /**
+     * @return a map from topic names to futures which can be used to check the status of
+     * individual deletions if the deleteTopics request used topic names. Otherwise return null.
+     */
+    public Map<String, KafkaFuture<Void>> topicNameValues() {
+        return nameFutures;
+    }
+
+    @Deprecated
```

Comment:
Helpful to include the version in which this has become deprecated. Also, the method should point the user to `topicNameValues`. For example:
```java
     * @deprecated Since 3.0 use {@link #topicNameValues} instead
```